chain/chain/src/chain.rs

```diff
@@ -952,13 +952,18 @@ impl Chain {
         chain_store_update.commit()?;
 
         // clear all trie data
-        let mut store_update = StoreUpdate::new_with_tries(self.runtime_adapter.get_tries());
         let stored_state = self.store().store().iter_prefix(ColState, &[]);
+
+        let mut keys: Vec<Vec<u8>> = vec![];
         for (key, _) in stored_state {
-            store_update.delete(ColState, key.as_ref());
+            keys.push(key.into());
         }
+
         let mut chain_store_update = self.mut_store().store_update();
-        chain_store_update.merge(store_update);
+        for key in keys {
+            chain_store_update.gc_col(ColState, &key)?;
+        }
+
```

Comment:
@bowenwang1996 there was some specific reason for creating `StoreUpdate::new_with_tries(self.runtime_adapter.get_tries())` and then `chain_store_update.merge(store_update)`, right? I guess it's in commit 1cdb346db7db35291f516ef2e70f2d070f0f4c9b. Does this change affect it?
If no, is it possible to optimize it to:
```
for (key, _) in stored_state {
    chain_store_update.gc_col(ColState, &key.into())?;
}
```
without saving `keys` explicitly?

Reply:
Yes it is necessary. Thank you @Kouprin for catching it! It potentially saves me 10+ hours of debugging :)

Reply:
So is my change okay? I think the pre collect step is because rust's single &mut reference restriction.

Reply:
I guess it's about to take `StoreUpdate::new_with_tries(self.runtime_adapter.get_tries())` and merge it into ChainStoreUpdate by `chain_store_update.merge(store_update)`. @bowenwang1996 is this correct approach?

Reply:
Yeah, looks i shouldn't deleted new_with_tries